proto/buf/alpha/registry/v1alpha1/recommendation.proto

```diff
@@ -53,3 +97,74 @@ message RecommendedTemplatesRequest {}
 message RecommendedTemplatesResponse {
   repeated RecommendedTemplate templates = 1;
 }
+
+message ListRecommendedRepositoriesRequest {
+  uint32 page_size = 1;
+  // The first page is returned if this is empty.
+  string page_token = 2;
+  bool reverse = 3;
+}
+
+message ListRecommendedRepositoriesResponse {
+  repeated RecommendedRepository repositories = 1;
+  // There are no more pages if this is empty.
+  string next_page_token = 2;
+}
+
+message ListRecommendedTemplatesRequest {
+  uint32 page_size = 1;
+  // The first page is returned if this is empty.
+  string page_token = 2;
+  bool reverse = 3;
+}
+
+message ListRecommendedTemplatesResponse {
+  repeated RecommendedTemplate templates = 1;
+  // There are no more pages if this is empty.
+  string next_page_token = 2;
+}
+
+message ListRecommendationsRequest {
+  ResourceType resource_type = 1;
+  uint32 page_size = 2;
+  // The first page is returned if this is empty.
+  string page_token = 3;
+  bool reverse = 4;
+}
+
+message ListRecommendationsResponse {
+  repeated Recommendation recommendations = 1;
+  // There are no more pages if this is empty.
+  string next_page_token = 2;
+}
+
+message AddRecommendationRequest {
+  Resource resource = 1;
+  string description = 2;
+}
+
+message AddRecommendationResponse {
+  Recommendation recommendation = 1;
+}
+
+message UpdateRecommendationRequest {
+  ResourceType resource_type = 1;
+  string id = 2;
+  string description = 3;
+}
+
+message UpdateRecommendationResponse {}
+
+message UpdateRecommendationsRanksRequest {
+  ResourceType resource_type = 1;
+  repeated string recommendation_ids = 2;
+}
+
+message UpdateRecommendationsRanksResponse {}
+
+message DeleteRecommendationRequest {
+  ResourceType resource_type = 1;
+  string id = 2;
```

Comment:
This shouldn't need the type, it should operate on the `id` alone, right?

Reply:
we can, but we will need to try to delete the record from both recommendation tables as we don't know where this actually lives. I think it might be better to let frontend include this information?

Reply:
This is an unfortunate consequence of an implementation detail (the two different tables). I don't think we should be exposing that to the API, we should make two deletes if we have to. What do you think?

Reply:
sounds good for me!